### ISSUE TITLE
Write a transition guide for asciitable => io.ascii migration

### ISSUE THREAD

**Issue by taldcroft** [astropy/astropy#846]
Provide guidance for people migrating their analysis and code from `asciitable` to `io.ascii`.  The biggest difference is between `Table` and `recarray`.

Kelle writes:

"from the bottom of my heart, I request that you post some useful operations like the one below to help people switch between the two datatypes in the transition phase. there's no way i would have been able to figure that out on my own."

Referring to:
BTW, if you really want an asciitable-like recarray (e.g. for legacy code), you can do:

```
>>> dat = io.ascii.read(...)
>>> dat_recarray = np.array(dat).view(np.recarray)
```

Basically that lets you access columns like dat_recarray.my_column instead of dat['my_column'].  But you lose a lot of Table goodies in that translation process.
